File: crates/weave-graph-parse/tests/fixtures/sample.zig
const std = @import("std");

const Greeter = struct {
    name: []const u8,

    pub fn greet(self: Greeter) []const u8 {
        return formatName(self.name);
    }
};

pub fn formatName(name: []const u8) []const u8 {
    return name;
}

File: crates/weave-graph-parse/tests/fixtures/sample.rs
use std::collections::HashMap;

pub struct Greeter {
    name: String,
}

impl Greeter {
    pub fn new(name: String) -> Self {
        Self { name }
    }

    pub fn greet(&self) -> String {
        format_name(&self.name)
    }
}

trait Farewell {
    fn bye(&self) -> String;
}

impl Farewell for Greeter {
    fn bye(&self) -> String {
        self.greet()
    }
}

fn format_name(name: &str) -> String {
    name.to_string()
}
